jdaviz/configs/cubeviz/plugins/slice/slice.py

```diff
@@ -145,6 +145,9 @@ def _update_data(self, x_all):
                 # leave in the pre-init state and don't update the wavelength/slice
                 return
 
+        # Also update unit when data is updated
+        self.wavelength_unit = x_all.unit.to_string()
```

Comment:
can this safely remove setting `wavelength_unit` in `_on_global_display_unit_changed` since that later calls this with the new quantity passed?

Reply:
Good call, I think you're right.